src/forms-pipe.js

```diff
@@ -121,6 +133,22 @@ export async function formsPipe(state, req) {
   const { path } = state.info;
   const resourcePath = `${path}.json`;
 
+  // verify captcha
+  const { fetch, config } = state;
+  const { 'captcha-secret-key': captchaSecretKey, 'captcha-type': captchaType } = config;
+  if (captchaType && captchaType !== 'reCaptcha v2') {
+    return error(log, `The captcha type ${captchaType} you have configured is not currently supported.`, 500, res);
+  }
+  if (captchaType && !captchaSecretKey) {
+    return error(log, 'You must configure ', 500, res);
+  }
+  if (captchaType) {
+    const captchaPassed = await verifyCaptcha(fetch, req.headers.get('x-google-captcha-token'), captchaSecretKey);
```

Comment:
I think the captcha response is included in the `g-recaptcha-response` form field, certainly not the header.